woocommerce/google-listings-and-ads
# Error notice if WooCommerce Admin isn't active

### Changes proposed in this Pull Request:

Closes #67.

This checks for WooCommerce Admin before registering the services used by Google Listings and Ads.

- It [checks to see if the `woocommerce_admin_disabled` filter is true](https://github.com/woocommerce/google-listings-and-ads/blob/c179de26004c9aa2c53cc93a044d8baf68b72e4a/src/WCAdminValidator.php#L24-L35) _before registering any GLA services_:
    https://github.com/woocommerce/google-listings-and-ads/blob/c179de26004c9aa2c53cc93a044d8baf68b72e4a/src/Infrastructure/GoogleListingsAndAdsPlugin.php#L107-L115
    This should catch a disabled WooCommerce Admin before activation, during plugin load, and before deactivation (these last two actions currently cause an error when WooCommerce Admin is disabled).
- Like to the PHP 7.3 version check, it outputs an error notice (in all of `/wp-admin/`) if WooCommerce Admin is discovered to be disabled:
    ![image](https://user-images.githubusercontent.com/228780/118840785-01e9c680-b8c8-11eb-9ab3-2237312d60ac.png)

#### Notes
- The WC Admin validation is run before registering GLA services, which happens during plugin activation, deactivation, and is [hooked to the `plugins_loaded` action](https://github.com/woocommerce/google-listings-and-ads/blob/c179de26004c9aa2c53cc93a044d8baf68b72e4a/src/Infrastructure/GoogleListingsAndAdsPlugin.php#L84-L90) if the plugin is already loaded.
- Some extensions, like WC Payments, [use `WC_ADMIN_VERSION_NUMBER`](https://github.com/Automattic/woocommerce-payments/blob/ef5c68b24d0d85773b0e10c9efc4cc7dc25b549e/includes/class-wc-payments.php#L381-L392) but it might not be available when the dependency checks are done during plugin load, or may be set even if WooCommerce admin is disabled.
- Some deactivation actions (like in the Notes) might be skipped if WooCommerce Admin is disabled after GLA was already enabled, but there's not much to do there since those actions require WC Admin.



### Detailed test instructions:

1. Deactivate GLA.
2. Install and activate this helper plugin to disable WooCommerce Admin using the hook: https://gist.github.com/joshuatf/006111b6f43c3ab3d2cad39e3443822b
     - Confirm it's working (the Marketing and Analytics sections don't show in the navigation menu, for example)
3. Activate GLA.
    - Confirm there are no errors in the log
    - Confirm that the error notice displays correctly on all pages of `/wp-admin/`
    - Confirm that the WC Admin sections don't show in the navigation menu (so the Google Listings and Ads link doesn't show either)
6. Deactivate GLA.
    - Confirm that there are no errors.
5. Deactivate the helper plugin and reactivate GLA.
    - Confirm there are no errors in the log.
    - Confirm that the error notice no longer appears on any `/wp-admin/` pages.
    - Confirm that the WC Admin sections – and the Google Listings and Ads link – are visible again in the navigation menu (and work correctly).

Bonus: Update the required PHP version to something higher than your version (e.g., `17.3`) in [`VersionValidator`](https://github.com/woocommerce/google-listings-and-ads/blob/4c6b4a51e9c7072f0c3f3fa36e23d0c1ac0351d3/src/Internal/Requirements/VersionValidator.php#L37-L41) and confirm that GLA doesn't load and an error notice is displayed.

### Changelog Note:

> Fix - Warn about disabled WooCommerce Admin.
